discord_rpc/DiscordRpc.hx

```diff
@@ -158,7 +158,13 @@ private extern class DiscordRpcExterns
         if (DiscordRpc.onSpectate != null) DiscordRpc.onSpectate(_secret);
     private static inline function _onRequest(_data : RawConstPointer<JoinRequest>) : Void {
         var star : cpp.Star<JoinRequest> = cast _data;
-        if (DiscordRpc.onRequest != null) DiscordRpc.onRequest(star);
+        var request:DiscordJoinRequest = {
+            userId: star.userId,
+            username: star.username,
```

Comment:
since we're cleaning up / getting things working I think this functions argument could be changed to `_data : cpp.ConstStar<JoinRequest>`  and then use that directly when creating the typedef since its automatically dereferenced. Can also get rid of the `star` var and casting.